src/openms/source/CHEMISTRY/ElementDB.cpp

```diff
@@ -574,17 +591,34 @@ namespace OpenMS
 
   void ElementDB::addElementToMaps_(const string& name, const string& symbol, const unsigned int an, const Element* e)
   {
-    names_[name] = e;
-    symbols_[symbol] = e;
-    atomic_numbers_[an] = e;
+    #pragma omp critical(OpenMS_ElementDB)
+    {
+      // overwrite existing element if it already exists
+      // find() has to be protected here in a parallel context
+      if (atomic_numbers_.find(an) != atomic_numbers_.end())
+      {
+        // in order to ensure that existing elements are still valid and memory
+        // addresses do not change, we have to modify the Element in place
+        // instead of replacing it.
+        const Element* const_ele = atomic_numbers_[an];
```

Comment:
this is a duplicated lookup.
The result of the find() from above could be cached to avoid extra work.

Reply:
adding elements should never be performance critical, if it is then something is clearly wrong